packages/dd-trace/src/plugins/util/ip_extractor.js

```diff
@@ -28,12 +27,23 @@ const privateCIDRs = [
   'fd00::/8'
 ]
 
-const privateIPMatcher = new BlockList()
+let privateIPMatcher
 
-for (const cidr of privateCIDRs) {
-  const [address, prefix] = cidr.split('/')
+try {
+  const { BlockList } = require('net')
+  privateIPMatcher = new BlockList()
 
-  privateIPMatcher.addSubnet(address, parseInt(prefix), net.isIPv6(address) ? 'ipv6' : 'ipv4')
+  for (const cidr of privateCIDRs) {
+    const [address, prefix] = cidr.split('/')
+
+    privateIPMatcher.addSubnet(address, parseInt(prefix), net.isIPv6(address) ? 'ipv6' : 'ipv4')
+  }
+} catch (e) {
+  // Fallback for older Node.js versions without BlockList
```

Comment:
I implemented a whole polyfill for blocklist back when I added this feature for this exact reason. I then removed it once we stopped supporting versions of node old enough to not have blocklist.
Optional chaining was added around the same time as blocklist in node, and you're not gonna write try catch for every optional chaining you see right ?
I'm thus very confused as to why you're trying to add this back.